core/connection.go

```diff
@@ -84,6 +90,55 @@ func CreateConnection(pathName string, src, dst *ProvableChain, to time.Duration
 	return nil
 }
 
+func checkConnectionCreateReady(src, dst *ProvableChain) (bool, error) {
+	srcID := src.Chain.Path().ConnectionID
+	dstID := dst.Chain.Path().ConnectionID
+
+	if srcID == "" && dstID == "" {
+		return true, nil
+	}
+
+	getState := func(pc *ProvableChain) (conntypes.State, error) {
+		if pc.Chain.Path().ConnectionID == "" {
+			return conntypes.UNINITIALIZED, nil
+		}
+
+		latestHeight, err := pc.LatestHeight()
+		if err != nil {
+			return conntypes.UNINITIALIZED, err
+		}
+		queryHeight := clienttypes.NewHeight(latestHeight.GetRevisionNumber(), 0)
```

Comment:
In most cases, `RevisionHeight == 0` means the genesis height.
In this case you should simply use the latest height.

Reply:
fixed